src/framework-ondata.js

```diff
@@ -52,17 +52,19 @@ module.exports = function(payload, socketIndex) {
          if ( ! _.isFunction( this.nodeInstance.wrapper.fRunUpdater ) ) { 
 
              // Don't have a wrapper fRunUpdater 
+             profiler.stopEvent(eventStart);
              throw Error( 'No wrapper fRunUpdater function found!  Cannot run updater.' );
 
          } else { 
-
              // TODO: revisit this logic later, per rdf-pipeline/noflo-rdf-pipeline#35
              // save data used to see if we have a different state after running updater
              clearStateData( vni.errorState ); // clear last error state
 
              // Save wrapper to make it accessible from the Promise
              var wrapper = this.nodeInstance.wrapper;
-             var updaterStartTime = Date.now();
+
+             profiler.stopEvent(eventStart); // done with general event processing 
+             var updateStart = profiler.startUpdate(); 
```

Comment:
This could be moved into the Promise constructor function (currently line 70 w/"Execute fRunUpdater"). This would ensure the timer does not start until the updater is actually executed. If the system is under heavy load there may be a delay between the event being fired and the updater being executed.